plugin/sdk_v2/shared/bus/types.py

```
@@ -0,0 +1,489 @@
+    def where_eq(self, field: str, value) -> "BusList[TBusRecord]":
+        return self.filter(lambda item: getattr(item, field, None) == value)
+
+    def where_contains(self, field: str, value: str) -> "BusList[TBusRecord]":
+        return self.filter(lambda item: value in str(getattr(item, field, "")))
+
+    def where_regex(self, field: str, pattern: str, *, strict: bool = True) -> "BusList[TBusRecord]":
+        import re
+        regex = re.compile(pattern)
+        filtered = self.filter(lambda item: bool(regex.search(str(getattr(item, field, "")))))
+        if strict and filtered.count() == 0:
+            raise ValueError(f"no records matched regex for field={field!r} pattern={pattern!r}")
```

Ruff (0.15.5)

[warning] Avoid specifying long messages outside the exception class

(TRY003)